api/v1/loadtest_types.go

```diff
@@ -208,12 +208,10 @@ type LoadTestSpec struct {
 	Scenarios []Scenario `json:"scenarios,omitempty"`
 
 	// Timeout provides the longest running time allowed for a LoadTest.
-	// +optional
-	TimeoutSeconds *int32 `json:"timeoutSeconds,omitempty"`
+	TimeoutSeconds *int32 `json:"timeoutSeconds"`
 
 	//TTL provide the longest time a LoadTest could live on the luster.
-	// +optional
-	TTLSeconds *int32 `json:"ttlSeconds,omitempty"`
+	TTLSeconds *int32 `json:"ttlSeconds"`
```

Comment:
I'm not sure these should be mandatory. I think our system should have sane defaults. These should not be pointers if they are required. The pointers convey optionality ([API Conventions: Optional vs. Required](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required)).

Reply:
I will fix the pointer type if we still want it mandatory, thank for you pointing out!

I think whether we want to have them mandatory or not is really depend on what we do if these values are not set. My understanding is we definitely need these values to perform garbage collection.  
1. If we are going to give them default values if not set, we could definitely make them optional, but we would need to implement them in cluster configuration and check if they are actually there. Moreover I think these two values are more of properties of load tests than the cluster. 
2. If we are going to reject the load test without these values in controller, we might just reject it before we get to scheduling.